pkg/crc/machine/bundle/metadata.go

```diff
@@ -281,3 +282,24 @@ func DownloadDefault(preset preset.Preset) error {
 
 	return nil
 }
+
+func Download(preset crcPreset.Preset, bundleURI string, isDefault bool) (string, error) {
+	if isDefault {
+		if preset == crcPreset.OpenShift {
+			return bundleURI, DownloadDefault(preset)
+		}
+		return bundleURI, image.PullBundle(preset, "")
+	}
+	switch {
+	case strings.HasPrefix(bundleURI, "http://"), strings.HasPrefix(bundleURI, "https://"):
+		return download.Download(bundleURI, constants.MachineCacheDir, 0644, nil)
+	case strings.HasPrefix(bundleURI, "docker://"):
+		if err := image.PullBundle(preset, strings.TrimPrefix(bundleURI, "docker:")); err != nil {
```

Comment:
@cfergeau doing it here because we are already validating if `docker://` is prefix, if we do it in the `PullBundle` then we again need to verify it. also `PullBundle` have defaultURI which doesn't contain `docker:` as prefix.

Reply:
code/API would look nicer if `image.PullBundle` directly accepted `docker://`  URL. Yes some code would likely be a bit redundant, but trimming `docker://` feels like a better fit in `PullBundle` rather than expecting all callers to do it themselves.

Reply:
Done